src/vs/workbench/contrib/chat/browser/chatAttachmentResolve.ts

```diff
@@ -232,3 +236,75 @@ function symbolId(resource: URI, range?: IRange): string {
 	}
 	return resource.fsPath + rangePart;
 }
+
+// --- NOTEBOOKS ---
+
+const NOTEBOOK_CELL_OUTPUT_MIME_TYPE_LIST_FOR_CHAT_CONST = [
+	'text/plain',
+	'text/html',
+	'application/vnd.code.notebook.error',
+	'application/vnd.code.notebook.stdout',
+	'application/x.notebook.stdout',
+	'application/x.notebook.stream',
+	'application/vnd.code.notebook.stderr',
+	'application/x.notebook.stderr',
+	'image/png',
+	'image/jpeg',
+	'image/svg',
+];
+
+export function resolveNotebookOutputAttachContext(data: NotebookCellOutputTransferData, editorService: IEditorService): IChatRequestVariableEntry[] {
+	const notebookEditor = getNotebookEditorFromEditorPane(editorService.activeEditorPane);
+	if (!notebookEditor) {
+		return [];
+	}
+
+	const outputViewModel = getOutputViewModelFromId(data.outputId, notebookEditor);
+	if (!outputViewModel) {
+		return [];
+	}
+
+	const mimeType = outputViewModel.pickedMimeType?.mimeType;
+	if (mimeType && NOTEBOOK_CELL_OUTPUT_MIME_TYPE_LIST_FOR_CHAT_CONST.includes(mimeType)) {
+
+		// get the cell index
```

Comment:
I wonder if this logic can be moved into notebook land as a utility.

Reply:
good point, since it's 1:1. Let me shift that over and it and the chat contrib can share